lib/generator/files.rb

```diff
@@ -0,0 +1,44 @@
+require 'pathname'
+
+module Generator
+  module Files
+    class ImmutableFile
+      attr_reader :filename, :repository_root
+      def initialize(filename:,repository_root: nil)
+        @filename = filename
+        @repository_root = repository_root
+      end
+
+      def to_s
+        File.read(filename) if File.exist?(filename)
+      end
+
+      def abbreviated_commit_hash
+        GitCommand.abbreviated_commit_hash(git_path, relative_filename)
+      end
+
+      private
+
+      def relative_filename
+        Pathname.new(filename).relative_path_from(Pathname.new(repository_root)).to_s
+      end
+
+      def git_path
+        File.join(repository_root, '.git')
+      end
+    end
+
+    class MutableFile < ImmutableFile
```

Comment:
Does inheritance here make sense?  Does this pass the is a? test for inheritance, a MutableFile is a ImmutableFile?